app/src/main/java/org/mozilla/tv/firefox/architecture/KillswitchLinearLayout.kt

```diff
@@ -30,11 +31,16 @@ class KillswitchLinearLayout : LinearLayout {
     constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
     constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
 
+    // This is the most recent visibility that client code attempted to set on
+    // this view, whether or not it was actually applied
     private var desiredVisibility = this.visibility
 
     private var isAllowedByCurrentExperiment: Boolean? = null
     private var allowedInLocales: KillswitchLocales? = null
 
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    var localeManager = LocaleManager.getInstance()
```

Comment:
nit: I think it'd be more correct & less fragile if this came from the service locator but that might be too much boilerplate.

Reply:
Agreed, but it'd also be a fairly sizable change, which I'm trying to pull back from doing all the time.